.tekton/tasks/task-publish-code-engine-deployable-mapping.yml

```diff
@@ -0,0 +1,141 @@
+        else
+          IBM_CLOUD_REGION=$(jq -r '.region_id' /artifacts/_toolchain.json | awk -F: '{print $3}')
+        fi
+
+        ibmcloud login -a $(params.ibmcloud-api) -r $IBM_CLOUD_REGION --apikey $IBM_CLOUD_API_KEY
+        ibmcloud target -g $(params.resource-group)
+        ibmcloud plugin install code-engine
+        ibmcloud ce proj select -n $(params.code-engine-project)
+
+        DEPLOYABLE_JSON=$(ibmcloud ce app get -n $(params.app-name) -o json)
+        APPLICATION_URL=$(echo $DEPLOYABLE_JSON | jq -j '.status.url')
+
+        # Get Code Engine app ID to use as deployable mapping ID
+        DEPLOYABLE_GUID=$(ibmcloud ce app get -n $(params.app-name) -o json | jq -j '.metadata.uid')
+
+        # Update deployable mapping
+        TOOLCHAIN_ENVIRONMENT=$(jq -j '.region_id' /artifacts/_toolchain.json | awk -F: '{print $2}')
+
+        if [ "$TOOLCHAIN_ENVIRONMENT" == "ys1" ]; then
+          TOOLCHAIN_ENVIRONMENT="dev."
+        else
+          TOOLCHAIN_ENVIRONMENT=""
+        fi
+
+        TOOLCHAIN_REGION=$(jq -j '.region_id' /artifacts/_toolchain.json | awk -F: '{print $3}')
+        OTC_API_DM_URL="https://otc-api.${TOOLCHAIN_REGION}.devops.${TOOLCHAIN_ENVIRONMENT}cloud.ibm.com/api/v1/toolchain_deployable_mappings"
+        IAM_TOKEN=$(ibmcloud iam oauth-tokens --output JSON | jq -j '.iam_token')
+
+        # Check if an existing mapping exist with the given deployable guid in toolchain
+        EXISTING_MAPPING=$(curl -H "Authorization: ${IAM_TOKEN}" "${OTC_API_DM_URL}?deployable_guid=${DEPLOYABLE_GUID}")
+        MAPPING_GUID=$(echo ${EXISTING_MAPPING} | jq -j '.items[0].mapping_guid')
+
+        if [ "${MAPPING_GUID}" != "null" ]; then
+          # Delete mapping
+          curl -X DELETE -H "Authorization: ${IAM_TOKEN}" "${OTC_API_DM_URL}/${MAPPING_GUID}"
+        fi
+
+        RESOURCE_GROUP_ID=$(ibmcloud target --output JSON | jq -j '.resource_group.guid')
+
+        CONSOLE_URL=$(ibmcloud ce app get -n $(params.app-name) | grep "Console URL:" | egrep -o 'https?://[^ ]+')
+
+        # Create deployable mapping
+        MAPPING_JSON=$(cat <<END
+        {
+          "deployable": {
+            "deployable_guid": "${DEPLOYABLE_GUID}",
+            "type": "code_engine",
+            "container": {
+              "guid": "${RESOURCE_GROUP_ID}",
+              "type": "resource_group_id"
+            },
+            "url": [
+              "${APPLICATION_URL}",
+              "${CONSOLE_URL}"
+            ]
+          },
+          "toolchain": {
+            "toolchain_guid": "$(jq -j '.toolchain_guid' /artifacts/_toolchain.json)",
+            "region_id": "$(jq -j '.region_id' /artifacts/_toolchain.json)"
+          },
+          "source": {
+            "type": "service_instance",
+            "source_guid": "${PIPELINE_ID}"
+          }
+        }
+        END
+        )
+
+        curl -X POST -H "Authorization: ${IAM_TOKEN}" -H "Content-Type: application/json" -d "${MAPPING_JSON}" "${OTC_API_DM_URL}"
```

Comment:
Do we need to check the status code in the response on this to confirm success? The Tekton-catalog task does this: https://github.com/open-toolchain/tekton-catalog/blob/master/toolchain/task-publish-deployable-mapping.yaml#L254

Reply:
Updating to catch HTTP status